app/controllers/tickets_controller.rb

```diff
@@ -1,22 +1,23 @@
 # frozen_string_literal: true
 
 class TicketsController < ApplicationController
+  before_action :authenticate_user!
+
   def index
-    @tickets = Ticket.all
+    authorize Ticket
+    @tickets = policy_scope(Ticket)
     render json: @tickets
   end
 
   def create
-    reservation = [] 
-    
-    @ticket = Ticket.new(ticket_params).each do 
+    @ticket = Ticket.new(ticket_params)
       if @ticket.save
-        reservation << @ticket
+        render json: @ticket, status: :created
       else
         render json: @tickets.errors, status: :unprocessable_entity
       end
-    end
-    render json: reservation, status: :created
+  
+    render json: ticket, status: :created
```

Comment:
So you should probably remove that, as this would override anything from your `if` statement above, meaning it would override that case when it was note saved as well